src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -187,9 +195,9 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
         /// <inheritdoc/>
         public IEnumerable<PeerAddress> SelectPeersForDiscovery(int peerCount)
         {
-            // Randomly order the list of peers and return the amount asked for.
-            var allPeers = this.peerAddresses.OrderBy(p => this.random.Next());
-            return allPeers.Select(p => p.Value).Take(1000);
+            var discoverable = this.peerAddresses.Values.Where(p => p.LastDiscoveredFrom < this.dateTimeProvider.GetUtcNow().AddHours(-PeerSelector.DiscoveryThresholdHours));
+            var allPeers = discoverable.Take(1000).ToList();
```

Comment:
Should we not first apply randomness and then take to top 1000?

Reply:
@dangershony Ah yes you right. Fixed...